@here/harp-mapview/lib/DecodedTileHelpers.ts

```diff
@@ -298,24 +300,37 @@ export function getBufferAttribute(attribute: BufferAttribute): THREE.BufferAttr
  * The default `three.js` object used with a specific technique.
  */
 export type ObjectConstructor = new (
-    geometry?: THREE.Geometry | THREE.BufferGeometry,
-    material?: THREE.Material | THREE.Material[]
+    geometry: THREE.Geometry | THREE.BufferGeometry,
+    material: THREE.Material | THREE.Material[]
 ) => THREE.Object3D;
 /**
  * Gets the default `three.js` object constructor associated with the given technique.
  *
  * @param technique The technique.
+ * @param elevationEnabled True if elevation is enabled, false otherwise.
  */
-export function getObjectConstructor(technique: Technique): ObjectConstructor | undefined {
+export function getObjectConstructor(
+    technique: Technique,
+    tile: Tile,
+    elevationEnabled: boolean
+): ObjectConstructor | undefined {
     if (technique.name === undefined) {
         return undefined;
     }
     switch (technique.name) {
         case "extruded-line":
         case "standard":
-        case "terrain":
         case "extruded-polygon":
         case "fill":
+            return elevationEnabled
```

Comment:
should we be able to enable the displaced mesh at run time? Why not always construct a DisplacedMesh, it would then remove the need for elevationEnabled

Reply:
yes, we can do it at runtime. When we enable elevation, we clear the tile cache, and the newly constructed objects will use DisplaceMesh.

Reply:
true, enabling elevation will anyway clear the cache, so it isn't like you can work around this.

Reply:
> true, enabling elevation will anyway clear the cache, so it isn't like you can work around this.
Sorry, I don't get it. What do you want to work around?

Reply:
My thinking was to enable the displacement at runtime, without having to redecode the tiles, but I agree, it was a dumb idea, because we anyway need to clear the tile cache (such runtime switching of decoding doesn't exist anyway).